mlflow/resource_mlflow_model.go

```diff
@@ -3,112 +3,76 @@ package mlflow
-	UserID               string         `json:"user_id,omitempty" tf:"computed"`
-	LatestVersions       []ModelVersion `json:"latest_versions,omitempty" tf:"computed"`
-	Description          string         `json:"description,omitempty"`
-	Tags                 []Tag          `json:"tags,omitempty"`
-	RegisteredModelID    string         `json:"id,omitempty" tf:"computed,alias:registered_model_id"`
-}
-
-// registeredModel defines response from GET API op
-type registeredModel struct {
-	RegisteredModelDatabricks Model `json:"registered_model_databricks"`
-}
-
-type ModelsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewModelsAPI(ctx context.Context, m any) ModelsAPI {
-	return ModelsAPI{m.(*common.DatabricksClient), ctx}
-}
-
-func (a ModelsAPI) Create(m *Model) error {
-	return a.client.Post(a.context, "/mlflow/registered-models/create", m, m)
-}
-
-func (a ModelsAPI) Read(name string) (*Model, error) {
-	var m registeredModel
-	err := a.client.Get(a.context, "/mlflow/databricks/registered-models/get", map[string]string{
-		"name": name,
-	}, &m)
-	if err != nil {
-		return nil, err
-	}
-	return &m.RegisteredModelDatabricks, nil
-}
-
-// Update the model entity
-func (a ModelsAPI) Update(m *Model) error {
-	return a.client.Patch(a.context, "/mlflow/registered-models/update", m)
-}
-
-// Delete removes the model by its name
-func (a ModelsAPI) Delete(name string) error {
-	return a.client.Delete(a.context, "/mlflow/registered-models/delete", map[string]string{
-		"name": name,
-	})
-}
-
 func ResourceMlflowModel() *schema.Resource {
 	s := common.StructToSchema(
-		Model{},
-		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			delete(m, "latest_versions")
-			return m
+		mlflow.RegisteredModel{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			delete(s, "latest_versions")
+			s["name"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			}
+			s["registered_model_id"] = &schema.Schema{ // TODO: is this proper aliasing 1
+				Type:     schema.TypeString,
+				Optional: true,
+			}
+			return s
 		})
 
 	return common.Resource{
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var m Model
-			common.DataToStructPointer(d, s, &m)
-			if err := NewModelsAPI(ctx, c).Create(&m); err != nil {
+			w, err := c.WorkspaceClient()
+			if err != nil {
 				return err
 			}
-			d.SetId(m.Name)
+			var req mlflow.CreateRegisteredModelRequest
+			common.DataToStructPointer(d, s, &req)
+			// TODO: resp is nil but the request is non-nil throughout entire stacktrace
+			_, err = w.RegisteredModels.Create(ctx, req)
+			if err != nil {
+				return err
+			}
+			d.Set("registered_model_id", req.Name) // TODO: is this proper aliasing 2
+			d.SetId(req.Name)                      // TODO: model is null - is it ok to set Id via request and not response?
```

Comment:
@nfx I need to set the Id, however `w.RegisteredModels.Create(ctx, req)` returns a RegisteredModel of `nil`.  Is it ok to use the request params to get the model name?